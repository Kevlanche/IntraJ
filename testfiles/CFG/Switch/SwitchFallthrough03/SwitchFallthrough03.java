public class SwitchFallthrough03 {
    public void test(int value) {
        switch (value) {
            case 1:
                int a = 10;
            case 2:
            case 3:
                int b = 20;
                System.out.println(a + b);
                break;
            default:
                System.out.println("default");
        }
    }
}