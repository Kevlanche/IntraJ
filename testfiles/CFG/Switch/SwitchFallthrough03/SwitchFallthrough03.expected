NodeName: Entry[13] Successors: [VarAccess[28]]
NodeName: VarAccess[28] Successors: [ConstCase[31]]
NodeName: ConstCase[31] Successors: [IntegerLiteral[41], ConstCase[42]]
NodeName: IntegerLiteral[41] Successors: [VariableDeclarator[38]]
NodeName: VariableDeclarator[38] Successors: [IntegerLiteral[54]]
NodeName: IntegerLiteral[54] Successors: [VariableDeclarator[51]]
NodeName: VariableDeclarator[51] Successors: [TypeAccess[58]]
NodeName: TypeAccess[58] Successors: [VarAccess[59]]
NodeName: VarAccess[59] Successors: [VarAccess[64]]
NodeName: VarAccess[64] Successors: [VarAccess[65]]
NodeName: VarAccess[65] Successors: [AddExpr[63]]
NodeName: AddExpr[63] Successors: [MethodAccess[60]]
NodeName: MethodAccess[60] Successors: [Exit[25]]
NodeName: Exit[25] Successors: []
NodeName: ConstCase[42] Successors: [IntegerLiteral[54], ConstCase[44]]
NodeName: ConstCase[44] Successors: [IntegerLiteral[54], DefaultCase[71]]
NodeName: DefaultCase[71] Successors: [TypeAccess[75]]
NodeName: TypeAccess[75] Successors: [VarAccess[76]]
NodeName: VarAccess[76] Successors: [StringLiteral[80]]
NodeName: StringLiteral[80] Successors: [MethodAccess[77]]
NodeName: MethodAccess[77] Successors: [Exit[25]]

NodeName: Exit[25] Predecessor: [MethodAccess[60], MethodAccess[77]]
NodeName: MethodAccess[60] Predecessor: [AddExpr[63]]
NodeName: AddExpr[63] Predecessor: [VarAccess[65]]
NodeName: VarAccess[65] Predecessor: [VarAccess[64]]
NodeName: VarAccess[64] Predecessor: [VarAccess[59]]
NodeName: VarAccess[59] Predecessor: [TypeAccess[58]]
NodeName: TypeAccess[58] Predecessor: [VariableDeclarator[51]]
NodeName: VariableDeclarator[51] Predecessor: [IntegerLiteral[54]]
NodeName: IntegerLiteral[54] Predecessor: [VariableDeclarator[38], ConstCase[42], ConstCase[44]]
NodeName: VariableDeclarator[38] Predecessor: [IntegerLiteral[41]]
NodeName: IntegerLiteral[41] Predecessor: [ConstCase[31]]
NodeName: ConstCase[31] Predecessor: [VarAccess[28]]
NodeName: VarAccess[28] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: ConstCase[42] Predecessor: [ConstCase[31]]
NodeName: ConstCase[44] Predecessor: [ConstCase[42]]
NodeName: MethodAccess[77] Predecessor: [StringLiteral[80]]
NodeName: StringLiteral[80] Predecessor: [VarAccess[76]]
NodeName: VarAccess[76] Predecessor: [TypeAccess[75]]
NodeName: TypeAccess[75] Predecessor: [DefaultCase[71]]
NodeName: DefaultCase[71] Predecessor: [ConstCase[44]]
